### PR TITLE
Support Service Hierarchy auto matching.

### PR DESCRIPTION
**Pull request by wankai123**
relate to [11669](https://github.com/apache/skywalking/discussions/11669)

- [ ] If this pull request closes/resolves/fixes an existing issue, replace the issue number. Closes #<issue number>.
- [X] Update the [`CHANGES` log](https://github.com/apache/skywalking/blob/master/docs/en/changes/changes.md).

### CODE REVIEW

**Review by mrproliu — APPROVED** [review]

File: oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/hierarchy/HierarchyService.java [hunk 1]
```diff
@@ -60,84 +67,149 @@ private Map<String, List<String>> getHierarchyDefinition() {
...
         }
+
+        buildInstanceHierarchyRelation(upperInstanceName, upperServiceName, upperServiceLayer, lowerInstanceName,
+                                           lowerServiceName, lowerServiceLayer);
+    }
+
+    public void startAutoMatchingServiceHierarchy() {
+        if (!this.isEnableHierarchy) {
+            return;
+        }
```
Comment (kezhenxu94) [inline oap-server/server-core/src/main/java/org/apache/skywalking/oap/server/core/hierarchy/HierarchyService.java:138]: Let's move this check to the provider `CoreModuleProvider` so we don't bother to initialize everything that is never used.

File: oap-server/server-starter/src/main/resources/hierarchy-definition.yml [hunk 2]
```diff
@@ -13,17 +13,34 @@
...
+    K8S_SERVICE: short-name
+
+  GENERAL:
+    K8S_SERVICE: lower-short-name-remove-ns
+
+  MYSQL:
+    K8S_SERVICE: lower-short-name-remove-ns
+
+  VIRTUAL_DATABASE:
+    MYSQL: ~
```
Comment (wu-sheng) [inline oap-server/server-starter/src/main/resources/hierarchy-definition.yml:35]: What does `~` mean?

Comment (wu-sheng) [reply oap-server/server-starter/src/main/resources/hierarchy-definition.yml:35]: It means there is no auto rule, right?

Comment (wankai123) [reply oap-server/server-starter/src/main/resources/hierarchy-definition.yml:35]: yes

**Review by wu-sheng — APPROVED** [review]

**Pull request merged**